src/pages/rekey-search/rekey-search.ts

```diff
@@ -1,19 +1,33 @@
 import { Subscription } from 'rxjs/Subscription';
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { IonicPage, NavController, NavParams, Platform, ViewController } from 'ionic-angular';
 import { BasePageComponent } from '../../shared/classes/base-page';
 import { AuthenticationProvider } from '../../providers/authentication/authentication';
-import { Store } from '@ngrx/store';
+import { Store, select } from '@ngrx/store';
 import { StoreModel } from '../../shared/models/store.model';
-import { RekeySearchViewDidEnter } from './rekey-search.actions';
+import { RekeySearchViewDidEnter, SearchBookedTest } from './rekey-search.actions';
+import { Observable } from 'rxjs/Observable';
+import { map } from 'rxjs/operators';
+import { getIsLoading, getHasSearched } from './rekey-search.selector';
+import { getRekeySearchState } from './rekey-search.reducer';
+
+interface RekeySearchPageState {
+  isLoading$: Observable<boolean>;
+  hasSearched$: Observable<boolean>;
+}
 
 @IonicPage()
 @Component({
   selector: 'page-rekey-search',
   templateUrl: 'rekey-search.html',
 })
-export class RekeySearchPage extends BasePageComponent {
+export class RekeySearchPage extends BasePageComponent implements OnInit {
+
+  pageState: RekeySearchPageState;
 
+  staffNumber: string = '';
+  applicationReference: string = '';
+  searchResults: any[] = [];
```

Comment:
Same again, could we use a more safe type?

Reply:
Added `TestSlot` type